pkg/generate/app/pipeline.go

```diff
@@ -43,7 +45,11 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
 	name, ok := NameSuggestions{source, input}.SuggestName()
 	if !ok {
-		name = fmt.Sprintf("app%d", rand.Intn(10000))
+		name = fmt.Sprintf("app-%04d", rand.Intn(10000))
+	}
+	if name == input.Name {
+		// Make output image name consistently different than input image name
+		name = namer.GetName(name, "app", v2.RepositoryNameTotalLengthMax)
```

Comment:
Might be over concerning, but if one of the two constants change in the future we could potentially be passing a negative number.
The question is, do we need to be super specific here? When we derive the name from a different source and don't get into that if clause, we won't be truncating long names.
Long names will be caught in the validation.